pkg/synchromanager/clustersynchro/cluster_synchro.go

```diff
@@ -406,12 +407,13 @@ func (s *ClusterSynchro) resourceSynchroRunner() {
 			}()
 
 			s.storageResourceSynchros.Range(func(_, value interface{}) bool {
-				go value.(*ResourceSynchro).Run(s.handlerStopCh)
+				s.synchroWaitGroup.StartWithChannel(s.handlerStopCh, value.(*ResourceSynchro).Run)
 				return true
 			})
 		}()
 
 		<-s.handlerStopCh
+		s.synchroWaitGroup.Wait()
```

Comment:
Maybe it would be simpler to call wait after the `s.closer` are closed. I don't think a temporary start/stop is necessary to wait. 
example:
```go
        for {
		select {
		case <-s.runResourceSynchroCh:
		case <-s.closer:
			break
		}
		// other code
	}
	s.synchroWaitGroup.Wait()
````